docs/07-faq/005-why-a-language.md

```diff
@@ -4,18 +4,79 @@ id: why-a-language
+
+Let's look at a simple example:
+
+```ts
+bring cloud;
+
+let queue = new cloud.Queue();
+let counter = new cloud.Counter();
+let bucket = new cloud.Bucket();
+
+queue.add_consumer(inflight (message: str) => {
+  let i = counter.inc();
+  bucket.put("file-${i}.txt", message);
+});
+```
+
+`cloud.Queue`, `cloud.Counter`, and `cloud.Bucket` are *preflight objects*. They represent cloud infrastructure resources. When compiled to a specific cloud provider, such as AWS, a Terraform file will be produced with the provider's implementation of these resources. The `queue.add_consumer()` method is a *preflight method* that configures the infrastructure to invoke a particular *inflight function* for each message in the queue.
+
+**Now comes the cool part**: the code that runs inside the inflight function interacts with the `counter` and the `bucket` objects through their *inflight methods* (`counter.inc()` and `bucket.put()`). These methods can only be called from inflight scopes.
+
+## Very cool, but what here cannot be done by a library or compiler extension?
+
+In existing languages, where there is no way to distinguish between multiple execution phases, it is impossible to naturally represent this idea that an object has methods that can only be executed from within a specific execution phase.
+
+In addition, other parts of Wing's *cloud-oriented* cannot be represented naturally without native support from the language. Wwe've seen some worthy efforts in projects like [Pulumi's Function Serialization](https://www.pulumi.com/docs/intro/concepts/function-serialization/) and [Functionless](https://functionless.org/). But although these efforts come close, they either have to make compromises in safety (e.g. it is possible to capture mutable objects) or in the programming model (e.g. type system is too complex).
+
+We believe that Wing, a language that natively supports this paradigm, will make using it much easier by streamlining common patterns, in a way that is impossible to accomplish with existing ones. Kind of like what C++ did for object orientation. You could write object-oriented code in C, but you had to work hard to make it work without native support from the language.
+
+## So how does writing code in Wing compare to other solutions?
+
+By natively supporting the cloud-oriented programming paradigm, Wing allows developers to write very few lines of code and let the compiler take care of IAM policies and other cloud mechanics.
+
+Below we've written the same simple demo app in Wing and other solutions. This simple app uses a Function to upload a text file to a Bucket, even without the `Counter` seen above that further complicates matters in other solutions:
+
+import CodeComparison from '../src/components/CodeComparison';
```

Comment:
What is this?

Reply:
Look at the screenshot in the PR description